andela/suyabay
# [#120500993] create endpoint that retrieves the data of the commenter

#### What does this PR do?
- Write an endpoint that retrieves all the data of the commenter of the comment of an episode.
- Write tests that test the endpoints in various cases.
#### Description of Task to be completed?
- Create a route `/v1/episodes/:name/comments/:comment_id/commenter` that calls the method `getEpisodeCommenter` in the `CommentController`
-  Create `getEpisodeCommenter` method to implement the commenter data retrieval
-  Create another method `displayCommenterData` that returns the data leveraging on the userTransformer
-  Use the userTransformer  to transform the user result
-  Write tests that assert that;
-  The episode does not exist if the comment_id passed is invalid for the episode with status code 404.
-  The data of the commenter of the comment under the episode are retrieved with status code 200.
#### How should this be manually tested?
- We need POSTMAN for this to be manually tested, POSTMAN is an extension plugin on google chrome.
-  To retrieve the data of the commenter 0f the comment under an episode,  send a `GET` request with this endpoint  `http://suyabay.app/api/v1/episodes/{name}/comments/{comment_id}/commenter`. If the {episode-name} is invalid, you should see a JSON response `The episode does not exist`, status code `404`. If the {episode_name} is valid, but the {comment_id} is invalid for the episode, you should see a JSON response `Comment not available for this episode, try another id`, status code `404`. If the {episode_name} and the {comment_id} is valid , you should get the data of the commenter of the comment under the  episode with status code`200`.
#### Any background context you want to provide?

None
#### What are the relevant pivotal tracker stories?
- #120500993
#### Screenshots (if appropriate)

![screen shot 2016-05-30 at 12 35 35 am](https://cloud.githubusercontent.com/assets/14159420/15636726/31f51006-2600-11e6-991d-d47f81cc7e39.png)

![screen shot 2016-05-30 at 12 35 50 am](https://cloud.githubusercontent.com/assets/14159420/15636727/31fe2f6a-2600-11e6-9d01-1a848dd5948f.png)
#### Questions:

[@unicodeveloper](https://github.com/unicodeveloper)
[@andela-oadebayo](https://github.com/andela-oadebayo)
